be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidator.scala

```diff
@@ -14,6 +14,37 @@ import scala.util.Success
 
 object MessageValidator extends ContentValidator with AskPatternConstants {
 
+  /** Extracts the useful parameters from a rpc message
+    *
+    * @param rpcMessage
+    *   rpc message which contains the parameters to extract
+    * @return
+    *   the decoded data, the id of the LAO, the PublicKey of the sender and the channel
+    */
+  def extractData[T](rpcMessage: JsonRpcRequest): (T, Hash, PublicKey, Channel) = {
+    val message: Message = rpcMessage.getParamsMessage.get
+    val data: T = message.decodedData.get.asInstanceOf[T]
+    val laoId: Hash = rpcMessage.extractLaoId
+    val sender: PublicKey = message.sender
+    val channel: Channel = rpcMessage.getParamsChannel
+
+    (data, laoId, sender, channel)
+  }
+
+  /** Runs the multiple checks of the validators
+    *
+    * @param list
+    *   List of checks which return a GraphMessage
+    * @return
+    *   GraphMessage: passes the rpcMessages to Left if successful right with pipeline error
+    */
+  def runChecks(list: List[GraphMessage]): GraphMessage = {
+    if (list.head.isLeft && !list.tail.isEmpty)
```

Comment:
I moved these functions at the top of the file. In my opinion, it seems more clear to have them before all the checks. 

I also added comments here and there, and changed `runList` to `runChecks` as suggested.